src/streamlit_app.py

```diff
@@ -103,27 +103,27 @@ def architecture_dialog():
             st.write(WELCOME)
 
     # draw_messages() expects an async iterator over messages
-    async def amessage_iter():
+    async def amessage_iter() -> AsyncGenerator[ChatMessage, None]:
         for m in messages:
             yield m
 
     await draw_messages(amessage_iter())
 
     # Generate new message if the user provided new input
-    if input := st.chat_input():
-        messages.append(ChatMessage(type="human", content=input))
-        st.chat_message("human").write(input)
+    if _input := st.chat_input():
```

Comment:
Why add an underscore here?

Reply:
[ruff A001](https://docs.astral.sh/ruff/rules/builtin-variable-shadowing/)

It's advisable to avoid using variable names that are the same as built-in variables, as this can sometimes lead to unexpected issues.
Of course, in this example, input() is not used, so there won't be any problems.

I personally think the simplest way to keep it straightforward is to add a leading underscore.
If you don't like underscores, you can change it to a more meaningful name, such as `user_input` or `chat_input`.

Reply:
Let's change it to `user_input` TY

Reply:
Finish